flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java

```diff
@@ -234,9 +301,23 @@ public OUT readRecord(OUT reuse, byte[] bytes, int offset, int numBytes) throws
 		}
 		
 		if (parseRecord(parsedValues, bytes, offset, numBytes)) {
-			// valid parse, map values into pact record
-			for (int i = 0; i < parsedValues.length; i++) {
-				reuse.setField(parsedValues[i], i);
+			if (pojoTypeClass == null) {
+				// result type is tuple
+				Tuple result = (Tuple) reuse;
+				for (int i = 0; i < parsedValues.length; i++) {
+					result.setField(parsedValues[i], i);
+				}
+			} else {
+				// result type is POJO
+				for (int i = 0; i < parsedValues.length; i++) {
+					try {
+						pojoFields[i].set(reuse, parsedValues[i]);
+					} catch (IllegalAccessException e) {
+						LOG.error("Cannot set value to given type object!", e);
```

Comment:
This error will not be caused by invalid data.
You should rethrow this exception to kill the job. If the field cannot be set, something is seriously wrong and the job should not continue.